### ISSUE TITLE
Unknown keyword: :redis_instance from RedisClient:: Config#initialize

### ISSUE THREAD

**Issue by snowfield702** [DataDog/dd-trace-rb#2371]
**Current behaviour**
I get the following error:
```
bundler: failed to load command: unicorn (/vendor/bundle/ruby/3.1.0/bin/unicorn)
/vendor/bundle/ruby/3.1.0/gems/redis-client-0.7.3/lib/redis_client/config.rb:20:in `initialize': unknown keyword: :redis_instance (ArgumentError)
	from /vendor/bundle/ruby/3.1.0/gems/redis-client-0.7.3/lib/redis_client/config.rb:156:in `initialize'
	from /vendor/bundle/ruby/3.1.0/gems/redis-client-0.7.3/lib/redis_client.rb:124:in `new'
	from /vendor/bundle/ruby/3.1.0/gems/redis-client-0.7.3/lib/redis_client.rb:124:in `config'
	from /vendor/bundle/ruby/3.1.0/gems/redis-5.0.2/lib/redis/client.rb:22:in `config'
	from /vendor/bundle/ruby/3.1.0/gems/redis-5.0.2/lib/redis.rb:157:in `initialize_client'
	from /vendor/bundle/ruby/3.1.0/gems/redis-5.0.2/lib/redis.rb:73:in `initialize'
	from /vendor/bundle/ruby/3.1.0/gems/ddtrace-1.6.0/lib/datadog/tracing/contrib/redis/patcher.rb:26:in `initialize'
```

It seems that the parameter added by Redis patcher #initialize is the cause.
https://github.com/DataDog/dd-trace-rb/blob/411449608c20be3b95e178b6c750bd542604fddb/lib/datadog/tracing/contrib/redis/patcher.rb#L24

**Expected behaviour**
I would like the error not to occur.

**Steps to reproduce**
Use the following gem versions:
- ddtrace 1.6.0
- redis-client v0.7.3
- redis 5.0.2

It happens when we run Redis.new(some_args).

**How does `ddtrace` help you?**

It is very useful for daily service monitoring.
**Environment**
* **ddtrace version: 1.6.0**
* **Configuration block (`Datadog.configure ...`):**
* **Ruby version: 3.1.0**
* **Operating system: Debian GNU/Linux 11**
* **Relevant library versions: N/A**

**Comment by TonyCTHsu** [comment]
@snowfield702 Thanks for reporting! I missed patching RedisClient for redis-rb 5. Will have a look

**Comment by TonyCTHsu** [comment]
👋 @snowfield702 , I have just looked into `redis-rb` 5 implementation, released from this August. It might take some time before I figure out a way to address this. 

It seems to me that previous versions of `ddtrace` would not raise exception, however, it also could not instrument `redis-rb` 5 properly. 

Could you confirm this behaviour from your experience running `redis-rb` 5 with previous version of `ddtrace`?

**Comment by snowfield702** [comment]
We will avoid using the redis instrument until this issue is resolved.
We look forward to the day we can use it again. 😉
```ruby
Datadog.configure do |c|
  # c.tracing.instrument :redis
end
```

**Comment by TonyCTHsu** [comment]
👋 @snowfield702 , I just released `1.6.1`. This will prevent the error by providing a version cap (We will not execute our redis patch). Sorry to cause the issue.

I am going close this issue for now and track our support for redis 5 in #2373
